utils/training.py

```diff
@@ -69,14 +71,17 @@ def supervised_training(
     n_iterations: int = int(1e5),
     check: bool = False,
     out_channels: int = 2,
-    augmentations:Optional[bool] = False,
-    eps:float = 1e-5, 
-    sigma: int = None, 
-    lower_bound: float = None, 
-    upper_bound: float = None, 
+    augmentations: Optional[bool] = False,
+    eps: float = 1e-5,
+    sigma: int = None,
+    lower_bound: float = None,
+    upper_bound: float = None,
     test_paths: Optional[Tuple[str]] = None,
     test_label_paths: Optional[Tuple[str]] = None,
     save_root: Optional[str] = None,
+    n_samples_train: Optional[int] = None,
+    n_samples_val: Optional[int] = None,
+    dataset_class=HeatmapDataset,
```

Comment:
I have made the dataset class more flexible, so that I can use this for my own training.